src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs

```diff
@@ -2647,7 +2647,7 @@ var projectUniqueNamesForBuildIntToUpdate
             if (updatedPackageSpec.RestoreSettings.HideWarningsAndErrors &&
                 !restoreResult.Result.Success)
             {
-                await MSBuildRestoreUtility.ReplayWarningsAndErrorsAsync(restoreResult.Result.LockFile, logger);
+                await MSBuildRestoreUtility.ReplayWarningsAndErrorsAsync(restoreResult.Result.LockFile?.LogMessages, logger);
```

Comment:
I guess my above comment about deprecating lock the assets file method applies perfectly :)